compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1369,11 +1369,13 @@ impl<'a> Parser<'a> {
             ))
         } else {
             // Field access `expr.f`
+            let span = lo.to(self.prev_token.span);
             if let Some(args) = seg.args {
-                self.dcx().emit_err(errors::FieldExpressionWithGeneric(args.span()));
+                self.dcx()
+                    .create_err(errors::FieldExpressionWithGeneric(args.span()))
+                    .stash(span, StashKey::GenericInFieldExpr);
```

Comment:
re: comment above

> Also, can you stash using the span of the field instead?

Then you can use `.stash(seg.ident.span` here instead. Using span logic (`.to`) sometimes plays weirdly with macros, so keeping the span simpler is better.